pkg/operator/operatorclient.go

```diff
@@ -97,19 +97,24 @@ func (c OperatorClient) GetOperatorStateWithQuorum(ctx context.Context) (*operat
 	return &instance.Spec.OperatorSpec, &instance.Status.OperatorStatus, instance.GetResourceVersion(), nil
 }
 
-func (c OperatorClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, applyConfiguration *applyoperatorv1.OperatorSpecApplyConfiguration) error {
-	desiredSpecApplyConf := &applyoperatorv1.StorageApplyConfiguration{
-		Spec: &applyoperatorv1.StorageSpecApplyConfiguration{OperatorSpecApplyConfiguration: *applyConfiguration},
-	}
+func (c OperatorClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, applySpecConfiguration *applyoperatorv1.OperatorSpecApplyConfiguration) error {
+	desiredSpecApplyConf := applyoperatorv1.Storage(globalConfigName).
+		WithSpec(&applyoperatorv1.StorageSpecApplyConfiguration{
+			OperatorSpecApplyConfiguration: *applySpecConfiguration})
 	_, err := c.Client.Storages().Apply(ctx, desiredSpecApplyConf, metav1.ApplyOptions{FieldManager: fieldManager})
 	return err
 }
 
-func (c OperatorClient) ApplyOperatorStatus(ctx context.Context, fieldManager string, applyConfiguration *applyoperatorv1.OperatorStatusApplyConfiguration) error {
-	desiredStatusApplyConf := &applyoperatorv1.StorageApplyConfiguration{
-		Status: &applyoperatorv1.StorageStatusApplyConfiguration{OperatorStatusApplyConfiguration: *applyConfiguration},
+func (c OperatorClient) ApplyOperatorStatus(ctx context.Context, fieldManager string, applyStatusConfiguration *applyoperatorv1.OperatorStatusApplyConfiguration) error {
+	for i := range applyStatusConfiguration.Conditions {
+		if applyStatusConfiguration.Conditions[i].LastTransitionTime.IsZero() {
+			lastTransitionTime := metav1.Now()
+			applyStatusConfiguration.Conditions[i].LastTransitionTime = &lastTransitionTime
+		}
 	}
```

Comment:
So, this is a new thing and we have to be careful about how we apply status. I am still trying to understand this, but most operators apply this differently - https://github.com/openshift/cluster-kube-descheduler-operator/blob/7c82e14b9b163731d5149c543573bdeecfadc8ea/pkg/operator/operatorclient/interfaces.go#L130

Reply:
Thanks for the info! Good to know.

Reply:
Though different operators apply this differently, I refer the `kube-descheduler-operator` logic refine our apply to make it more solid.